rust/crates/libs/edlcodegen-macros/src/utils.rs

```diff
@@ -0,0 +1,182 @@
+    } else if name == "Box" {
+        return handle_boxed_to_boxed(expr, path, field);
+    }
+
+    // If we get here then we're looking at a plain unwrapped type T
+    // in the edl struct.
+
+    // Called when the edl type is T and the target type is Box<T>
+    if field.attr == TargetFieldHint::Boxed {
+        return handle_unboxed_to_boxed(expr, field);
+    }
+
+    quote! { #expr.into() }
+}
+
+/// Converts `Option<T>`, `Vec<T>` generically using a provided closure pattern.
+fn convert_generic<F>(
+    _expr: TokenStream2,
+    path: &Path,
+    field: &FieldInfo,
+    wrapper: F,
+) -> TokenStream2
+where
+    F: Fn(TokenStream2) -> TokenStream2,
+{
+    let inner_ty = extract_inner_type(path);
+    let inner_expr = generate_conversion(quote! { inner }, &inner_ty, field);
+    wrapper(inner_expr)
+}
+
+/// `Option<T> <-> Option<U>`
+fn handle_option_to_option(expr: TokenStream2, path: &Path, field: &FieldInfo) -> TokenStream2 {
+    convert_generic(expr.clone(), path, field, |inner_expr| {
+        quote! { #expr.map(|inner| { #inner_expr }) }
+    })
+}
+
+/// `Vec<T> <-> Vec<U>`
+fn handle_vec_to_vec(expr: TokenStream2, path: &Path, field: &FieldInfo) -> TokenStream2 {
+    convert_generic(expr.clone(), path, field, |inner_expr| {
+        quote! { #expr.into_iter().map(|inner| { #inner_expr }).collect() }
+    })
+}
+
+/// `Box<T> <-> Box<U>`
+fn handle_boxed_to_boxed(expr: TokenStream2, path: &Path, field: &FieldInfo) -> TokenStream2 {
+    convert_generic(expr.clone(), path, field, |inner_expr| {
+        quote! {{let inner = *#expr; Box::new(#inner_expr) }}
+    })
+}
+
+/// Extracts the inner type T from  `Option<T>`, `Vec<T>` and `Box<T>`
+fn extract_inner_type(path: &Path) -> Type {
+    let Some(seg) = path.segments.first() else {
+        panic!("empty path");
+    };
+
+    let PathArguments::AngleBracketed(args) = &seg.arguments else {
+        panic!("expected angle-bracketed generic args");
+    };
+
+    let Some(syn::GenericArgument::Type(typ)) = args.args.first() else {
+        panic!("expected one generic argument");
+    };
+
+    typ.clone()
+}
+
+fn handle_unboxed_to_boxed(expr: TokenStream2, field: &FieldInfo) -> TokenStream2 {
+    match field.dir {
+        Direction::ToTarget => quote! { Box::new(#expr.into()) },
+        Direction::FromTarget => quote! { (*#expr).into() },
+    }
+}
+
+fn handle_array_to_vec(expr: TokenStream2, field: &FieldInfo) -> TokenStream2 {
+    match field.dir {
+        // edl type: [T; N] → target: Vec<T>
+        Direction::ToTarget => {
+            quote! { #expr.iter().map(|x| x.clone().into()).collect::<Vec<_>>() }
+        }
+
+        // Target: Vec<T> → edl type: [T; N]
+        Direction::FromTarget => {
+            quote! {{
+                core::array::from_fn(|i| {
+                    if i < #expr.len() {
+                        #expr[i].clone().into()
+                    } else {
+                        Default::default()
```

Comment:
>Default::default() [](http://example.com/codeflow?start=24&length=18)

How could this happen?

If developer-type array is going from Host -> |fb| -> Enclave

Does this mean somehow the array on the Enclave side got created too big, and now it will be filled with empty entries?

Reply:
Should we error out of `#expr.len() != N`  (where does N come from, lol)